packages/vats/src/core/types.js

```diff
@@ -99,9 +99,11 @@
 
 /**
  * @template T
+ * @template [U={}]
+ * @template [V={}]
```

Comment:
now that there are three slots, consider giving them meaningful names.
```suggestion
 * @template B Bidirectional
 * @template [C={}] Consume only
 * @template [P={}] Produce only
```
(and `K in keyof` below)

Reply:
Done